lib/nexthop_group.h

```diff
@@ -43,6 +43,10 @@ struct nexthop_group *nexthop_group_new(void);
 void nexthop_group_delete(struct nexthop_group **nhg);
 
 void nexthop_add(struct nexthop **target, struct nexthop *nexthop);
+void nexthop_group_add_sorted(struct nexthop_group *nhg,
```

Comment:
I'm slightly concerned about the API design here.  Nexthop groups are supposed to be immutable objects; once created they don't change.  I would rather have some kind of a `nexthop_group_create(nexthops)` which makes it abundantly clear that you supply all the nexthops and get a group in return, which isn't to be touched afterwards.

I assume you're using this in the construction of nexthop groups, prior to them being "immutabilized"?

Reply:
Per you and @donaldsharp 's discussion in the meeting a couple weeks back, I added the nexhtop group manipulation functions to a `private` header.